src/main/java/com/google/cloud/spanner/pgadapter/parsers/DateParser.java

```diff
@@ -114,13 +116,28 @@ protected byte[] binaryParse() {
     if (this.item == null) {
       return null;
     }
-    LocalDate localDate =
-        LocalDate.of(this.item.getYear(), this.item.getMonth(), this.item.getDayOfMonth());
+    return convertToPG(this.item);
+  }
+
+  static byte[] convertToPG(Date value) {
+    LocalDate localDate = LocalDate.of(value.getYear(), value.getMonth(), value.getDayOfMonth());
     long days = localDate.toEpochDay() - PG_EPOCH_DAYS;
     int daysAsInt = validateRange(days);
     return IntegerParser.binaryParse(daysAsInt);
   }
 
+  public static byte[] convertToPG(ResultSet resultSet, int position, DataFormat format) {
+    switch (format) {
+      case SPANNER:
```

Comment:
Why haven't we added the conversion logic for Cloud Spanner format for some of the data types?

Reply:
In Java, case statements automatically fall through, so here the same logic is used for both when the format is `SPANNER` and `POSTGRESQL_TEXT`. This is different from some other programming languages where this would mean that there is no conversion for `SPANNER`.